test/linalg.jl

```diff
@@ -62,3 +64,57 @@ end
         end
     end
 end
+
+###### Test jacobian matrix ######
+params(A::SparseMatrixCSC) = params(A.nzval)
+fitparams(A::SparseMatrixCSC, p) = SparseMatrixCSC(A.m, A.n, A.colptr, A.rowval, fitparams(A.nzval, p))
```

Comment:
It is a function to load parameters to match the target type. We have `params` and `fitparams` because ForwardDiff can only take real vectors as input parameters, however, we want to test complex numbers too.